### PR TITLE
[Select] Apply design language to Select

### PR DESCRIPTION
**Pull request by dleroux**
**Design still pending, no need to review until [WIP] is removed**

select

### WHY are these changes introduced?

partly addresses: https://github.com/Shopify/polaris-ux/issues/363

### WHAT is this pull request doing?

Applying design language to Select. Current Select shouldn't be affected.

I'm also removing the Design Language modification from Backdrop. Once we roll this out, we'll be able to delete that mixin.

The icons have not been updated. My understanding is that these will be updated when icons get updated.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by dleroux** [comment]
@martenbjork can you ping me here once the design is approved?

**Comment by dleroux** [comment]
In speaking with @danrosenthal we decided to remove the `wip` and adjust if the design changes.

File: documentation/Color system.md [hunk 1]
```diff
@@ -43,19 +43,18 @@ The surface role is used for the backgrounds of the UI. With a light color schem
```
Comment (dleroux) [inline documentation/Color system.md]: Rebasing, not sure why this has changed.

Comment (dleroux) [reply documentation/Color system.md]: We'll hold on shipping this. @danrosenthal will fix this in another PR.

**Comment by dleroux** [comment]
I had removed the Backdrop because it was only useful for the checkbox in the new design language. However, because OptionList has a copy of the checkbox I decided to bring it back until we get the opportunity to remove the OptionList from inside the Autocomplete (the reason why additional props were needed on the fork checkbox).

File: src/components/Select/Select.scss [hunk 2]
```diff
@@ -17,32 +17,30 @@ $stacking-order: (
...
-  // stylelint-disable-next-line selector-max-specificity
-  .Input:focus ~ .Backdrop {
-    @include control-backdrop(focused-error);
+    opacity: var(--override-one, 0.4);
+    @include recolor-icon(var(--p-icon-disabled, color('ink', 'lighter')));
+  }
+
+  &.globalTheming {
+    .Backdrop {
+      border-color: var(--p-border-secondary-disabled);
```
Comment (tmlayton) [inline src/components/Select/Select.scss]: Both secondary and onSurface variants are used for borders around inputs. This should be consistent, one or the other. I lean towards onSurface.

Comment (tmlayton) [reply src/components/Select/Select.scss]: /cc @danrosenthal

Comment (danrosenthal) [reply src/components/Select/Select.scss]: All interactive elements should use the secondary borders, that creates a separation between the decorative (on surface) borders, and functional borders so we can ensure color contrast on interactive elements. We should fix any uses of on surface borders in interactive elements

**Pull request merged**